src/irr/asset/CGLIWriter.cpp

```diff
@@ -153,12 +153,28 @@ namespace irr
 			{
 				const auto ptrBeginningOfRegion = data + region->bufferOffset;
 				const auto layerSize = getFullSizeOfLayer(region->imageSubresource.mipLevel);
```

Comment:
wrong thing to use, you should be computing that from `region` in its entirety (cause size of layer depends on `bufferRowLength` and `bufferImageHeight` and pixel byte size and eventual depth) also you can compute that without the mip-level at all.

P.S. Check Vulkan Specificationm for semantics of `bufferRowLength` and `bufferImageHeight` when block compressed textures are involved (does the length and height still mean pixels or blocks?)

Reply:
it's computed like bellow

> wrong thing to use, you should be computing that from region in its entirety (cause size of layer depends on bufferRowLength and bufferImageHeight and pixel byte size and eventual depth) also you can compute that without the mip-level at all.

Reply:
I will check it out